website/addons/wiki/static/pagedown-ace/Markdown.Editor.js

```diff
@@ -210,7 +216,293 @@
             that.uiManager = uiManager;
         };
 
-    }
+    };
+
+    var getExtension = /(?:\.([^.]+))?$/;
+    var validImgExtensions = ['jpg', 'jpeg', 'png', 'gif', 'bmp'];
+
+    var localFileHandler = function(files, cm, init, fixupInputArea) {
+        var multiple = files.length > 1;
+        var urls = [];
+        var num = cm.addLinkDef(init) + 1;
+        var deferred = [];

```

Comment:
This is a misnomer because `deferred` is not a `$.Deferred`; it's an Array of promises. How about `promises`?